src/environments/toytext/blackjack.jl

```diff
@@ -0,0 +1,100 @@
+export BlackjackEnv
+
+
+# 1 = Ace, 2-10 = Number cards, Jack/Queen/King = 10
+deck = vcat(collect(1:10), [10, 10, 10])
+
+@inline draw_card(seed::AbstractRNG) = rand(seed, deck)
+@inline draw_hand(seed::AbstractRNG) = [draw_card(seed), draw_card(seed)]
+@inline usable_ace(hand) = 1 ∈ hand && sum(hand) + 10 ≤ 21
+@inline sum_hand(hand) = usable_ace(hand) ? sum(hand) + 10 : sum(hand)
+@inline is_bust(hand) = sum_hand(hand) > 21
+@inline score(hand) = is_bust(hand) ? 0 : sum_hand(hand)
+@inline isnatural(hand) = sort(hand) == [1, 10]
+
+mutable struct BlackjackEnv <: AbstractEnv
+    action_space::DiscreteSpace{UnitRange{Int64}}
+    observation_space::VectSpace{Tuple{DiscreteSpace{UnitRange{Int64}},DiscreteSpace{UnitRange{Int64}},DiscreteSpace{UnitRange{Int64}}}}
+    dealer::Array
+    player::Array
+    state::Tuple{Int64,Int64,Int64}
+    action::Int
+    reward::Int
+    done::Bool
+    natural::Bool
+    rng::AbstractRNG
+end
+
+"""Simple blackjack environment
+Blackjack is a card game where the goal is to obtain cards that sum to as
+near as possible to 21 without going over.  They're playing against a fixed
+dealer.
+Face cards (Jack, Queen, King) have point value 10.
+Aces can either count as 11 or 1, and it's called 'usable' at 11.
+This game is placed with an infinite deck (or with replacement).
+The game starts with each (player and dealer) having one face up and one
+face down card.
+The player can request additional cards (hit=1) until they decide to stop
+(stick=0) or exceed 21 (bust).
+After the player sticks, the dealer reveals their facedown card, and draws
+until their sum is 17 or greater.  If the dealer goes bust the player wins.
+If neither player nor dealer busts, the outcome (win, lose, draw) is
+decided by whose sum is closer to 21.  The reward for winning is +1,
+drawing is 0, and losing is -1.
+The observation of a 3-tuple of: the players current sum,
+the dealer's one showing card (1-10 where 1 is ace),
+and whether or not the player holds a usable ace (0 or 1).
+This environment corresponds to the version of the blackjack problem
+described in Example 5.1 in Reinforcement Learning: An Introduction
+by Sutton and Barto.
+http://incompleteideas.net/book/the-book-2nd.html
+"""
+
+function BlackjackEnv(;natural=false,rng = Random.GLOBAL_RNG)
+    # natural = flag to payout on a "natural" blackjack win, like casion rules
+    action_space = DiscreteSpace(2)
+    observation_space = VectSpace((DiscreteSpace(32),DiscreteSpace(11),DiscreteSpace(2)))
```

Comment:
Could you use MultiDiscreteSpace here?
The type will be more concise.

Reply:
Alright I'll change this to MultiDiscreteSpace